hermes-client/src/main/java/pl/allegro/tech/hermes/client/webclient/WebClientHermesSender.java

```diff
@@ -0,0 +1,79 @@
+package pl.allegro.tech.hermes.client.webclient;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import pl.allegro.tech.hermes.client.HermesMessage;
+import pl.allegro.tech.hermes.client.HermesResponse;
+import pl.allegro.tech.hermes.client.HermesSender;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toMap;
+import static pl.allegro.tech.hermes.client.HermesResponseBuilder.hermesResponse;
+
+public class WebClientHermesSender implements HermesSender {
+
+    private final WebClient webClient;
+
+    public WebClientHermesSender(WebClient webClient) {
+        this.webClient = webClient;
+    }
+
+    @Override
+    public CompletableFuture<HermesResponse> send(URI uri, HermesMessage message) {
+        final Mono<ClientResponse> clientResponse = webClient.post()
+                .uri(uri)
+                .syncBody(message.getBody())
+                .headers(httpHeaders -> httpHeaders.setAll(message.getHeaders()))
+                .exchange();
+
+        return clientResponse
```

Comment:
This could probably be shorten to:
```java
clientRespponse.flatMap(response -> response
                        .bodyToMono(String.class)
                        .map(body -> hermesResponse()
                                .withBody(body)
                                .withHttpStatus(response.rawStatusCode())
                                ...))
                .toFuture();
```

Reply:
plus we need something like:

```
.switchIfEmpty(NO_BODY)
```

where
```
 private final static Mono<String> NO_BODY = Mono.just("");
```

Map will not be called if there's no body. 
You should check how frontends behave, but I guess they send no body on 2xx response.